webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -438,18 +455,47 @@ final case class ProjectsResponderADMLive(
    *
    * @param id  the project's identifier (IRI / shortcode / shortname / UUID)
    *
-   * @return [[ProjectRestrictedViewSettingsGetResponseADM]]
+   * @return [[ProjectRestrictedViewSettingsResponseADM]]
    */
   override def projectRestrictedViewSettingsGetRequestADM(
     id: ProjectIdentifierADM
-  ): Task[ProjectRestrictedViewSettingsGetResponseADM] =
+  ): Task[ProjectRestrictedViewSettingsResponseADM] =
     projectRestrictedViewSettingsGetADM(id)
       .flatMap(ZIO.fromOption(_))
       .mapBoth(
         _ => NotFoundException(s"Project '${getId(id)}' not found."),
-        ProjectRestrictedViewSettingsGetResponseADM
+        ProjectRestrictedViewSettingsResponseADM
       )
 
+  /**
+   * Sets project's restricted view settings.
+   *
+   * @param iri the project's iri,
+   * @param user  requesting user,
+   * @param size  value to be set,
+   * @return [[ProjectRestrictedViewSettingsResponseADM]].
+   */
+  override def setProjectRestrictedViewSettings(
+    iri: Iri.ProjectIri,
+    user: UserADM,
+    size: RestrictedViewSize
+  ): Task[ProjectRestrictedViewSizeResponseADM] = {
+    val id = ProjectIdentifierADM.IriIdentifier(iri)
+    for {
+      _ <- getProjectFromCacheOrTriplestore(id)
+             .flatMap(ZIO.fromOption(_))
+             .orElseFail(NotFoundException(s"Project '${getId(id)}' not found."))
+      _ <- ZIO
+             .fail(ForbiddenException("User is neither system nor project admin."))
+             .unless(user.permissions.isSystemAdmin || user.permissions.isProjectAdmin(iri.value))
```

Comment:
Maybe it would make sense to not have this method in the responder but add it to the `ProjectADMService` instead, wdyt?

Reply:
This actually make sense, because responder method looks like repetition to the one in service with additional triplestore query handling. Only one thing is that I will need to call this method also in another place in responder when the default value will be decided (the TODO I left). This will sets the value while creating new project. So it's feasible, but would require importing `ProjectADMRestService` into responder. This shouldn't make any concerns, right?

Reply:
The responder should not have a dependency on `ProjectADMRestService`. Actually we want to refactor the code so that we can remove the `Responders` entirely. 

The goal here is separating the concerns. The `ProjectADMService` knows nothing about authorization and when called it trusts that the caller has done its part. That is, you should move the method to `ProjectADMService` (without Rest) and inject this service into the responder. The responder currently handles authorization by itself and can reuse the service to perform the business logic.

Reply:
See also [our discussion on refactoring](https://www.notion.so/draft-RFC-0001-Package-Structure-Webapi-Refactoring-V3-56eee56e3e6e4d569ace273e611b696b?pvs=4)

Reply:
Done in https://github.com/dasch-swiss/dsp-api/pull/2794/commits/43b5a0e6183ef96b19eb588dd825f693f2f113e3